tests/test_diff_tables.py

```diff
@@ -239,6 +240,43 @@ def test_diff_sorted_by_key(self):
         self.assertEqual(expected, diff)
 
 
+class TestStringKeys(TestWithConnection):
+    def setUp(self):
+        super().setUp()
+
+        queries = [
+            "DROP TABLE IF EXISTS a",
+            "DROP TABLE IF EXISTS b",
+            "CREATE TABLE a(id varchar(100), comment varchar(1000))",
+            "COMMIT",
+        ]
+        for i in range(100):
+            queries.append(f"INSERT INTO a VALUES ('{uuid.uuid1(i)}', '{i}')")
+
+        queries += [
+            "COMMIT",
+            "CREATE TABLE b AS SELECT * FROM a",
+            "COMMIT",
+        ]
+
+        self.new_uuid = uuid.uuid1(32132131)
+        queries.append(f"INSERT INTO a VALUES ('{self.new_uuid}', 'This one is different')")
+
+        # TODO test unexpected values?
+        # queries.append(f"INSERT INTO a VALUES ('unexpected', '<-- this bad value should not break us')")
+
+        for query in queries:
+            self.connection.query(query, None)
+
+        self.a = TableSegment(self.connection, ("a",), "id", "comment")
+        self.b = TableSegment(self.connection, ("b",), "id", "comment")
+
+    def test_string_keys(self):
+        differ = TableDiffer()
+        diff = list(differ.diff_tables(self.a, self.b))
+        self.assertEqual(diff, [("-", (str(self.new_uuid), "This one is different"))])
```

Comment:
Some more tests here would be great, e.g.:

* Some rows are UUIDS, others are not
* Invalid or weird UUIDs (is that possible?)
* Maybe also some where there's a few more differences and no differences?

Reply:
Right now, if I see anything that's not UUID, I just throw an error. But sure, I can test that.